backend/src/test/java/com/github/gadenko/gamereviewsite/backend/controller/GameReviewControllerTest.java

```diff
@@ -219,4 +219,100 @@ void deleteGameReview_deleteAGameReview() {
                 //THEN
                 .expectStatus().is2xxSuccessful();
     }
+
+    @Test
+    void updateGameReviewById_whenValid_thenReturnUpdatedReview() {
+        //Given
+        GameReview gameReview = GameReview
+                .builder()
+                .title("TES5 Skyrim")
+                .headline("Hält der Titel was er verspricht?")
+                .gameDescription("Was für eine Fantasy Welt")
+                .picture("https://upload.wikimedia.org/wikipedia/en/2/26/X4_Foundations_Steam_Cover_Art.jpg")
+                .build();
+        GameReview addedGameReview = testClient.post()
+                .uri("http://localhost:" + port + "/api/gamereview")
+                .bodyValue(gameReview)
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody(GameReview.class)
+                .returnResult()
+                .getResponseBody();
+        //When
+        assertNotNull(addedGameReview);
+        GameReview updatedGameReview = GameReview
+                .builder()
+                .id(addedGameReview.getId())
+                .title("X4: Foundations")
+                .headline("Hält der Titel was er verspricht?")
+                .gameDescription("Schöner Weltraum Titel. Sehr viele möglickeiten der entfaltung im Space.")
+                .picture("https://upload.wikimedia.org/wikipedia/en/2/26/X4_Foundations_Steam_Cover_Art.jpg")
+                .build();
+        GameReview actual = testClient.put()
+                .uri("http://localhost:" + port + "/api/gamereview")
+                .bodyValue(updatedGameReview)
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody(GameReview.class)
+                .returnResult()
+                .getResponseBody();
+        //Then
+        GameReview expected = GameReview
+                .builder()
+                .id(addedGameReview.getId())
+                .title("X4: Foundations")
+                .headline("Hält der Titel was er verspricht?")
+                .gameDescription("Schöner Weltraum Titel. Sehr viele möglickeiten der entfaltung im Space.")
+                .picture("https://upload.wikimedia.org/wikipedia/en/2/26/X4_Foundations_Steam_Cover_Art.jpg")
+                .build();
+        assertEquals(expected,actual);
+    }
+
+    @Test
+    void updateGameReviewById_whenIdDoesNotExist_thenReturnNewReview(){
```

Comment:
Hier bin ich nicht ganz sicher, ob ich die Methode richtig verstanden habe - 
Wenn ein Update geschickt wird mit einer Id, die noch nicht in der Datenbank existiert, wird der neue Beitrag der Datenbank hinzugefügt, wenn ich es richtig verstanden habe (save-Methode von der MongoDB). 
Hier meine Fragen dazu: 

- Kann es überhaupt dazu kommen, dass eine nicht-existente Id mitgegeben wird?
- Wenn ja, wäre es dann nicht besser, den Nutzer zu informieren, dass ein neuer Eintrag angelegt wurde?
- Kann die DB dem Eintrag eine DB-Id verpassen, wenn der neue Eintrag schon eine Id hat?